### DISCUSSION TITLE
trying to change the background of the splash screen of android applications

### DISCUSSION THREAD

**Discussion by josquinet** [airsdk/Adobe-Runtime-Support#2634] [Q&A] [answered]
Before reporting this issue:

- you should consider whether this would be a question better suited to discussions above ^^ ;
- make sure you have searched for similar issues and add your experience there if one exists.

## Problem Description

Describe your problem in detail. Include the following information:

Good morning. I am making an App using Animate (last updated version) and Harmann's Air-Sdk (version 50.2.2.5). As of Android 2012, a splash screen is enabled in all applications. This link refers to that screen:
https://developer.android.com/guide/topics/ui/splash-screen?hl=es-419

My problem is that this splash screen has a black background, and my logo is very dark, so I would like to make this background white. I am trying to change it to white. For this I have done the following:

I create an animation or still image in Animate that has the desired background.
I export the animation or image as a PNG or JPEG file.
I copy the PNG or JPEG file into the "res/drawable" folder of my Adobe Air project.
I open the "AndroidManifest.xml" file of my Adobe Air project and add the following code inside the "application" tag:
php
`<activity android:name="com.adobe.air.ScreenOrientationPortrait" android:theme="@android:style/Theme.NoTitleBar.Fullscreen" android:screenOrientation="portrait">
      <intent-filter>
          <action android:name="android.intent.action.MAIN" />
          <category android:name="android.intent.category.LAUNCHER" />
      </intent-filter>
      <meta-data android:name="android.app.splash_screen_drawable" android:resource="@drawable/my_file"/>
</activity>`

when I try to create the aab file through animate (I've done it many times already with good results) it returns the error that I send in the attached image:
![segujndo error](https://user-images.githubusercontent.com/85651150/235418063-5da0eacf-175c-48a1-be67-a58a30b65c39.png)

Thanks for the help

**Comment by marchbold** [comment]
You need to add the `android:exported` attribute to your activity. This is a new security requirement for certain activities and services. 

```xml
<activity android:exported="false"
android:name="com.adobe.air.ScreenOrientationPortrait" android:theme="@android:style/Theme.NoTitleBar.Fullscreen" android:screenOrientation="portrait"> 
    <intent-filter> 
        <action android:name="android.intent.action.MAIN" /> 
        <category android:name="android.intent.category.LAUNCHER" /> 
    </intent-filter> 
    <meta-data android:name="android.app.splash_screen_drawable" android:resource="@drawable/my_file"/> 
</activity>
```

**Reply by josquinet** [reply]
Hello, I have added that line and it gives me the error of the attached image. I send my manifest additions in case there are any typos. Thank you so much:

`-<android>
-<manifestAdditions>
-<![CDATA[<manifest>
<uses-permission android:name="android.permission.INTERNET"/>
<activity android:exported="false"
android:name="com.adobe.air.ScreenOrientationPortrait" android:theme="@android:style/Theme.NoTitleBar.Fullscreen" android:screenOrientation="portrait"> 
    <intent-filter> 
        <action android:name="android.intent.action.MAIN" /> 
        <category android:name="android.intent.category.LAUNCHER" /> 
    </intent-filter> 
    <meta-data android:name="android.app.splash_screen_drawable" android:resource="@drawable/fondo_android"/> 
</activity>
</manifest>]]>
</manifestAdditions>
</android>`

**Reply by marchbold** [reply]
The activity node must go inside your application node:

```xml
<manifest> 
    <uses-permission android:name="android.permission.INTERNET"/> 

    <application>
        <activity android:exported="false" android:name="com.adobe.air.ScreenOrientationPortrait" android:theme="@android:style/Theme.NoTitleBar.Fullscreen" android:screenOrientation="portrait"> 
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />  
                <category android:name="android.intent.category.LAUNCHER" />  
            </intent-filter> 
            <meta-data android:name="android.app.splash_screen_drawable" android:resource="@drawable/fondo_android"/>  
        </activity> 
    </application>
</manifest>
```

**Comment by josquinet** [comment]
Here I attach the attached image of the comment of the previous intervention.
![tercer error fondo](https://user-images.githubusercontent.com/85651150/235630678-72918184-f59a-4895-b3a6-56aaf4d05640.png)

**Comment by josquinet** [comment]
hello!. I have tried putting the <application> node. it does not work. That node is there before and I think I'm improperly duplicating it. I send in text fil my entire manifest file and an image of the error that appears now when putting node <application>
I wanted to say that I have the android sdk installed on an external hard drive (just like the android program) and I don't know if that could be the cause of the error. thank you so much.

[comentario git hub.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/11409806/comentario.git.hub.txt)


![Captura de pantalla 2023-05-05 220741](https://user-images.githubusercontent.com/85651150/236560654-cf590a5b-dd7c-4665-873f-22b980f8707e.png)

**Reply by marchbold** [reply, marked as answer]
If you are trying to change the splash screen, not sure you are allowed to override an existing activity. You likely have to customise the theme:

https://github.com/airsdk/Adobe-Runtime-Support/issues/1503#issuecomment-1016008239

**Reply by josquinet** [reply]
Thanks I'll try

**Reply by josquinet** [reply]
https://github.com/airsdk/Adobe-Runtime-Support/issues/1503

here is the solution...
